### PR TITLE
[Doc] Use sphinx_book_theme

### PR DESCRIPTION
**Pull request by simon-mo**
## Why are these changes needed?
This PR updates the theme for use [Sphinx Book Theme](https://sphinx-book-theme.readthedocs.io/). A full rendering can be found [here](https://anyscale-ray-doc-assets.s3-us-west-2.amazonaws.com/index.html)

Note:
- Breadcrumbs are gone.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
  - Running few sanity checks:
     - [x] Google Analytics still works
     - [x] Read the docs build passes. It should appear in https://simon-ray.readthedocs.io/en/doc-new-theme/
     - [x]  Make sure version warnings still appear.

### CODE REVIEW

File: doc/source/conf.py [hunk 1]
```diff
@@ -213,33 +213,38 @@
...
 
 # Theme options are theme-specific and customize the look and feel of a theme
 # further.  For a list of options available for each theme, see the
 # documentation.
-#html_theme_options = {}
+html_theme_options = {
+    "repository_url": "https://github.com/ray-project/ray",
+    "use_repository_button": True,
+    "use_issues_button": True,
+    "use_edit_page_button": True,
```
Comment (simon-mo) [inline doc/source/conf.py:225]: These should replace our custom "open an issue" button.

Comment (richardliaw) [reply doc/source/conf.py:225]: nice >:)

**Comment by simon-mo** [comment]
Note: We need to wait for https://github.com/executablebooks/sphinx-book-theme/pull/198 to be merged and released.

**Comment by ericl** [comment]
image
image

@simon-mo several of the rllib tables look funny in the new theme, are there are fixes you can make to the config to make them look more compact?

**Comment by richardliaw** [comment]
I think the fix there is to do it via the scss; could look into it later this weekend.

**Review by bllchmbrs — APPROVED** [review]

**Review by richardliaw — COMMENTED** [review]
TODO before merge:

1. fix the tables as mentioned by eric
2. make sure centering works

We may want to just override the scss file with the corresponding changes for 1.0.

**Comment by simon-mo** [comment]
@richardliaw @ericl I updated the centering and table. 
- Full Preview: https://anyscale-ray-doc-assets.s3-us-west-2.amazonaws.com/index.html

### Image Centering
image

### Table Compact
image

**Review by richardliaw — APPROVED** [review]

**Comment by richardliaw** [comment]
image
last nit here @simon-mo

**Pull request merged**